electrum/gui/qml/qetxfinalizer.py

```diff
@@ -868,3 +871,101 @@ def update(self):
     @pyqtSlot(result=str)
     def getNewTx(self):
         return str(self._new_tx)
+
+
+class QETxSweepFinalizer(QETxFinalizer):
+    _logger = get_logger(__name__)
+
+    txinsRetrieved = pyqtSignal()
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        self._private_keys = ''
+        self._txins = None
+        self._amount = QEAmount(is_max=True)
+
+        self.txinsRetrieved.connect(self.update)
+
+    privateKeysChanged = pyqtSignal()
+    @pyqtProperty(str, notify=privateKeysChanged)
+    def privateKeys(self):
+        return self._private_keys
+
+    @privateKeys.setter
+    def privateKeys(self, private_keys):
+        if self._private_keys != private_keys:
+            self._private_keys = private_keys
+            self.update_privkeys()
+            self.privateKeysChanged.emit()
+
+    def make_sweep_tx(self):
+        address = self._wallet.wallet.get_unused_address()  # TODO: dont fail
```

Comment:
so let's just use `get_receiving_address` instead, to resolve that todo? :)

Reply:
> so let's just use `get_receiving_address` instead, to resolve that todo? :)

for desktop, it is implemented with number of fallbacks:

https://github.com/spesmilo/electrum/blob/8c9263fb200fc45f0b74def98dd6c1df9f2af9bd/electrum/gui/qt/main_window.py#L2388-L2394

Reply:
All that complexity is only there because of that "Get wallet address" button, which gives you a "new" address every time you click it. (not sure why we have that button tbh)
If you only want a single address, it is much easier, you can just call `wallet.get_receiving_address()`
![pic1](https://github.com/user-attachments/assets/24e4de0e-801e-43c4-bd47-d398fcb818a4)

Reply:
ack